### DISCUSSION TITLE
Pi5 network disconnects

### DISCUSSION THREAD

**Discussion by noiselevelzero** [rdz-oss/BattyBirdNET-Pi#46] [General]
I have been running Batty Birdnet off of a Pi5 for a few weeks now, and it certainly runs better than it did on a Pi4, except for one thing.  For a reason I cannot diagnose, BBN will eventually drop off the network, and will not again be visible until I reboot the PI5.  This almost always happen overnight, after the system becomes active collecting, and not during the day when it is not processing.  As far as I know, this never happened on the Pi4, but this consistently happens on a PI5. 

So far, I have tried multiple SD cards and even an NVME SSD, and have swapped Pi5's as well, none of which helped.  I have also tried a number of "keep alive" Network Manager strings set up in various ways to autoexec and keep the WIFI connected, none of which worked.

To make a final check, I wiped the SD card and rebuilt it with just the OS, but did not then install Batty Birdnet from the script.  I placed the card back in the pi, and reconnected my audiomoth so I could duplicate the same audio interface in case that was also an issue.  

So far, the Pi5 has been stable for several days, leading me to believe this issue is with the Batty Birdnet build in some way.  I could try to stress test the Pi (for instance with something like cpburn) but before I bother with that, does anyone else have an issue with their Pi5 using Batty Birdnet?  I really would like to have this working on the Pi5 reliably.

Thanks.

**Comment by noiselevelzero** [comment]
It's a week now with no network disconnects of the Pi5.  I will now reinstall Batty Birdnet from the script and advise the results.

**Reply by noiselevelzero** [reply]
Reinstalled from script, and the device has dropped off the network but not stopped observing as far as I can tell since the first night with BattyBirdnet loaded on the Pi5.  The symptoms are the main web page and an SSH connection is unreachable from anywhere on the network until a hard restart.

It is possible this is something completely unique and common to two separate PI5's, two separate Audiomoth devices, 4 power supplies, 3 USB cables, 4 SD cards, and an NVME drive.  However, I strongly suspect that this system is not Pi5 compatible.  :)  Lack of any replies makes me believe this is not an issue for others, or no one else is running on a Pi5.

Regards.

**Reply by rabede** [reply]
I don't know wether it has something to do with your problem and wether it helps somebody to find the cause. I have exactly the same problem on a different setting since a couple of days. 
I had https://github.com/Nachtzuster/BirdNET-Pi running on a Raspberry 4 flawlessly since february. Now it disconnects from local Wifi-Network short time after reboot. The recording keeps running, but the webinterface is not reachable. Maybe it has something to do with an update on a source both projects still share?

**Reply by rdz-oss** [reply]
Hi, I have the same issue with one of my setups on a Pi 4. It did not go away with a fresh install. Might be an issue of the current RaspberryPi OS.

**Reply by rdz-oss** [reply]
After a recent update of RaspberryOS it works better on my system that was faulty. You can try 

sudo apt-get update && sudo apt-get upgrade -y

in the shell (via ssh login or via UI in Tools - WebTerminal log in with the username (not birdnet) and passwort for the operating system that you defined when flashing the RaspberryOS.
Hope that works for you,

BR
Richard

**Comment by noiselevelzero** [comment]
Thank you, I'll give that a try.  I had been initiating a reboot every 4 hours to keep the system interface from becoming unavailable.  I'd also replaced the regular SD Card with an industrial level card, from the perspective that maybe the number of writes was causing early card failure.  Hopefully the latest version of the Raspberry Pi firmware does the trick.

**Reply by rdz-oss** [reply]
FYI I am also testing these options here: https://support.thepihut.com/hc/en-us/articles/360014695877-Raspberry-Pi-won-t-connect-to-WiFi

- Localization settings
- power supply 
etc.

**Comment by rdz-oss** [comment]
I found something in an IT journal that I am trying now to stabilize WIFI connection. It does appear to be a RaspberryOS issue: https://www.heise.de/ratgeber/Raspi-mit-Debian-verliert-Internet-Verbindung-9998575.html

It appears that a daemon is killing the wlan connection. It can be disabled by 

sudo systemctl disable avahi-daemon.socket
sudo systemctl disable avahi-daemon

I am checking the long term effects still. Has been ok for a few hours now.

**Comment by noiselevelzero** [comment]
Thanks.  I will try that.  You also turn off Wifi power management (which is on by default) using instructions here:
https://www.thedigitalpictureframe.com/stay-connected-enhancing-raspberry-pi-wi-fi-stability-by-turning-off-power-management/

**Comment by rdz-oss** [comment]
I have a new router and tried a machine (Pi 4) with a fresh OS install. Working fine for two days now on WIFI. Perhaps the issue has been resolved on OS level.

**Comment by rabede** [comment]
I tried all your tricks from above and it got a bit better, but still sometimes WIFI went down. Now I have cronjob to reboot the Pi every morning and no problems since.

**Comment by noiselevelzero** [comment]
Same here.   No matter what I do, the device still drops from wifi, and requires a reboot.  It is the only always-on Pi5 of the 3 I have that does that.   Yes, I have spared it with 2 other Pi5’s with the same outcome.

Otherwise, enjoying seeing bats return from Torpor here.  I spared out the Pi PS with an Astron linear supply, and a huge difference in Power Supply noise and false detections.

Rabede, how does your Cron job read?   Just a basic reboot?

Thanks,


Bob


On Apr 2, 2025, at 4:38 AM, rabede ***@***.***> wrote:

﻿

I tried all your tricks from above and it got a bit better, but still sometimes WIFI went down. Now I have cronjob to reboot the Pi every morning and no problems since.

**Reply by tphakala** [reply]
If you run "iwconfig wlan0" on RPI Linux shell, what does output look like?

**Reply by noiselevelzero** [reply]
Good link quality, and Wifi power management is "off".

Cheers.

**Reply by rabede** [reply]
@noiselevelzero 
yes, just a simple:

`0 7 * * * /sbin/reboot`

**Reply by noiselevelzero** [reply]
@rabede Thank you.

**Comment by noiselevelzero** [comment]
Thanks.  You are right, simple, but unless I catch it during a reboot, the main page is always available now.  14 species since they came out of torpor.

Diff question:  is it just me, or is anyone else unable to listen to non-shifted audio?  Sometimes I want to identify a non-bat noise, but cannot listen unshifted.

Thanks

Bob
Sent from my iPad

On Apr 2, 2025, at 1:52 PM, rabede ***@***.***> wrote:

﻿

@noiselevelzero<https://github.com/noiselevelzero>
yes, just a simple:

0 7 * * * /sbin/reboot

**Reply by rdz-oss** [reply]
Hi Bob,

audio replay works in regular speed on my current dev system. Still sometimes looses WIFI connection sometimes (I do not use the reboot mechanism) but less often. 
14 species! That is great! I managed to record 4 so far this spring. You must have a great location for bats.
BR
Richard

**Reply by noiselevelzero** [reply]
Hi, I do not seem to be able to get it to work, unless I shift the audio downward.  I can listen to non-shifted streams, but not listen to any recordings that way.  I cannot figure that out yet.

Yes, I am up to 14 species now.  My region of Virginia is what I call a seasonal jungle, with plenty of insects.  The plants grow at an enormous rate in the spring, the insects arrive early, in abundance, and the Bats do, too.

Regards and thanks for this wonderful resource for Bat monitoring,

Bob

P.s.: have you ever given thought to an addition to detect feeding calls by themselves and highlight them?
